src/app/credExplorer/basicPagerank.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+
+import stringify from "json-stable-stringify";
+import * as tf from "@tensorflow/tfjs-core";
+
+import {Graph} from "../../core/graph";
+
+/**
+ * Run eigenvector centrality, and print results to console.
+ */
+export default function basicPagerank(graph: Graph<any, any>): void {
```

Comment:
This function should return the stationary distribution (I'm fine for it to also print to console as a side effect)

Reply:
I disagree. In general, a function that computes and returns a value should not have effectful output. (Some people call this "command–query separation.") If it did, then callers who just want the value would be forced to put up with the noise. Choosing not to return the value avoids this problem by making there not be any such callers.

I intended for this module to be an exit point: you feed it data and get nothing out. I could later refactor it to export two functions—one query to find the stationary distribution from a graph, and one command to describe the output to the console. This might be helpful for your UI work—would you like me to do that in the next PR?

Reply:
Yep, I'd like the module to export the data for usage in the UI.
I view the print-to-console as a short-term hack UI, so IMO that should not live in basicPageRank - it should be implemented in the cred explorer app, as it is stand in for a real UI that would be generated in that file.